pkg/provisioner/lvm.go

```diff
@@ -0,0 +1,309 @@
+package provisioner
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	diskv1 "github.com/harvester/node-disk-manager/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/node-disk-manager/pkg/block"
+	"github.com/harvester/node-disk-manager/pkg/utils"
+)
+
+type LVMProvisioner struct {
+	name      string
+	vgName    string
+	blockInfo block.Info
+	device    *diskv1.BlockDevice
+
+	semaphoreObj *Semaphore
+}
+
+func NewLVMProvisioner(vgName string, device *diskv1.BlockDevice, blockInfo block.Info, semaphoreObj *Semaphore) *LVMProvisioner {
```

Comment:
Since these provisioners only have private variables (LVM & Longhorn), and you also have the interface here. You could do following changes:

1. Convert `LVMProvisioner` `LonghornV1Provisioner` to private.
2. Use interface as return in `NewLVMProvisioner` `NewLHV1Provisioner`

For example:

```suggestion
func NewLVMProvisioner(vgName string, device *diskv1.BlockDevice, blockInfo block.Info, semaphoreObj *Semaphore) Provisioner {
```

Reply:
good suggestion, updated!